addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/hw/HwSerialBridgeHandler.java

```
@@ -0,0 +1,275 @@
+
+        if (!this.getBridge().getStatus().equals(ThingStatus.ONLINE)) {
+            logger.warn("HomeWorks Bridge is offline and cannot update time on HomeWorks processor.");
+            if (updateTimeJob != null) {
+                updateTimeJob.cancel(false);
+                updateTimeJob = null;
+            }
+            return;
+        }
+
+        sendCommand("SD, " + dateString);
+        sendCommand("ST, " + timeString);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.error("Unexpected command for HomeWorks Bridge: {} - {}", channelUID, command);
+    }
+
+    private void handleIncomingMessage(String line) {
+        if (line == null || line.isEmpty()) {
+            return;
+        }
+
+        logger.info("Received message from HomeWorks processor: {}", line);
+        String[] data = line.replaceAll("\\s", "").toUpperCase().split(",");
+        if ("DL".equals(data[0])) {
+            try {
+                String address = data[1];
+                Integer level = Integer.parseInt(data[2]);
+                HwDimmerHandler handler = findHandler(address);
+                if (handler == null) {
+                    discService.declareUnknownDimmer(address);
+                } else {
+                    handler.handleLevelChange(level);
+                }
+            } catch (final Exception e) {
+                logger.error("Error parsing incoming message", e);
+            }
+        }
+
+    }
+
+    private HwDimmerHandler findHandler(String address) {
+        for (Thing thing : getThing().getThings()) {
+            if (thing.getHandler() instanceof HwDimmerHandler) {
+                HwDimmerHandler handler = (HwDimmerHandler) thing.getHandler();
+                if (address.equals(handler.getAddress())) {
+                    return handler;
+                }
+            }
+        }
+        return null;
```

Can the caller handle this or should you just throw an exception instead?